### PR TITLE
[Metric]Java metric api enhancement

### PR DESCRIPTION
**Pull request by ashione**
make tagkey transparent for upper level users

## Why are these changes needed?

Java users can pass map<string,string> tags for every metrics, so tagkey is transparent to them.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ashione** [comment]
@zhongchun @jovany-wang

**Review by zhongchun — COMMENTED** [review]
Adding some tests will be better.

File: java/runtime/src/main/java/io/ray/runtime/metric/Count.java [hunk 1]
```diff
@@ -22,6 +22,10 @@ public Count(String name, String description, String unit, Map<TagKey, String> t
     Preconditions.checkState(metricNativePointer != 0, "Count native pointer must not be 0.");
   }
 
+  public Count(String name, String description, Map<String, String> tags) {
```
Comment (jovany-wang) [inline java/runtime/src/main/java/io/ray/runtime/metric/Count.java:25]: Should we remove  https://github.com/ray-project/ray/pull/17811/files#diff-2ffdfba93c661a2ab4ac6a1b23ef5b4a971acb8bf7ac020c6a88fcdc48098fd8R13 ?

Btw if we remove them, this PR will be an API change. Maybe we can mark the old API as deprecated?

File: java/runtime/src/main/java/io/ray/runtime/metric/TagKey.java [hunk 2]
```diff
@@ -16,6 +18,20 @@ public String getTagKey() {
...
 
+  /**
+   * Convert pair of string and string map to tags map that can be recognized in native layer.
+   *
+   * @param tags metrics key value map.
+   * @return
+   */
+  public static Map<TagKey, String> tagsFromMap(Map<String, String> tags) {
+    Map<TagKey, String> tagKeyMap = new HashMap<>();
+    for (Map.Entry<String, String> entry : tags.entrySet()) {
```
Comment (jovany-wang) [inline java/runtime/src/main/java/io/ray/runtime/metric/TagKey.java:29]: tags.foreach() or tags.map()?

Comment (ashione) [reply java/runtime/src/main/java/io/ray/runtime/metric/TagKey.java:29]: foreach is just sugar formula.

File: java/runtime/src/main/java/io/ray/runtime/metric/Count.java (hunk 1 above)
Comment (ashione) [reply java/runtime/src/main/java/io/ray/runtime/metric/Count.java:25]: sounds great. I will add this annotation later.

**Review by jovany-wang — APPROVED** [review]

**Pull request merged**